subprojects/packagefiles/pcre/meson.build

```diff
@@ -80,24 +149,22 @@ char_tables = custom_target('chartables',
   output : 'pcre_chartables.c',
   command : [dftables, '@OUTPUT@'])
 
-pcre8 = build_target('pcre',
+pcre8 = library('pcre',
   pcre8_sources, char_tables,
-  c_args : compile_args,
-  target_type : libtype,
-  include_directories : conf_inc)
+  c_args: compile_args,
```

Comment:
Shouldn't you pass all deps you lookup above?

Reply:
Yeah probably. The interesting thing is that these dependencies seem to be used not by the library but utilities that it comes with.

Reply:
Do we build those utilities? If not, then we don't need to lookup for those deps at all.

Reply:
pcretest is built, yes.

I guess since they're all optional they can be removed.

Reply:
Yeah:

```
~/p/pcre-8.45 (master)> git grep SUPPORT_LIBZ *.c
pcregrep.c:#ifdef SUPPORT_LIBZ
pcregrep.c:#ifdef SUPPORT_LIBZ
pcregrep.c:#if defined SUPPORT_LIBZ || defined SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBZ
pcregrep.c:#ifdef SUPPORT_LIBZ
pcregrep.c:#ifdef SUPPORT_LIBZ
pcregrep.c:#ifdef SUPPORT_LIBZ
pcregrep.c:#if defined SUPPORT_LIBZ || defined SUPPORT_LIBBZ2
pcregrep.c:#if defined SUPPORT_LIBZ || defined SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBZ
pcregrep.c:#ifdef SUPPORT_LIBZ
~/p/pcre-8.45 (master)> git grep SUPPORT_LIBB *.c
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#if defined SUPPORT_LIBZ || defined SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#if defined SUPPORT_LIBZ || defined SUPPORT_LIBBZ2
pcregrep.c:#if defined SUPPORT_LIBZ || defined SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
pcregrep.c:#ifdef SUPPORT_LIBBZ2
~/p/pcre-8.45 (master)> git grep SUPPORT_LIBE *.c
pcretest.c:#if defined(SUPPORT_LIBREADLINE) || defined(SUPPORT_LIBEDIT)
pcretest.c:#if defined(SUPPORT_LIBREADLINE) || defined(SUPPORT_LIBEDIT)
pcretest.c:#if defined(SUPPORT_LIBREADLINE) || defined(SUPPORT_LIBEDIT)
mangix@TrapNAS ~/p/pcre-8.45 (master)> git grep SUPPORT_LIBR *.c
pcretest.c:#if defined(SUPPORT_LIBREADLINE) || defined(SUPPORT_LIBEDIT)
pcretest.c:#if defined(SUPPORT_LIBREADLINE)
pcretest.c:#if defined(SUPPORT_LIBREADLINE) || defined(SUPPORT_LIBEDIT)
pcretest.c:#if defined(SUPPORT_LIBREADLINE) || defined(SUPPORT_LIBEDIT)
```

It's all pcretest and pcregrep. The former is built but not the latter.

Reply:
ok good